src/libraries/Common/tests/System/IO/ByteLoggingStream.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO
+{
+    internal sealed class BytesLoggingStream : Stream
+    {
+        private readonly Stream _stream;
+        private readonly Action<bool, string> _log;
+
+        public BytesLoggingStream(Stream stream, Action<bool, string> log)
+        {
+            _stream = stream;
+            _log = log;
+        }
+
+        public override bool CanRead => _stream.CanRead;
+        public override bool CanSeek => _stream.CanSeek;
+        public override bool CanWrite => _stream.CanWrite;
+        public override long Length => _stream.Length;
+        public override long Position { get => _stream.Position; set => _stream.Position = value; }
+        public override void Flush() => _stream.Flush();
+        public override long Seek(long offset, SeekOrigin origin) => _stream.Seek(offset, origin);
+        public override void SetLength(long value) => _stream.SetLength(value);
+        public override int ReadTimeout { get => _stream.ReadTimeout; set => _stream.ReadTimeout = value; }
+        public override int WriteTimeout { get => _stream.WriteTimeout; set => _stream.WriteTimeout = value; }
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _stream.Dispose();
+            }
+        }
+
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            int read = _stream.Read(buffer, offset, count);
+            FormatBytes(read: true, buffer.AsSpan(offset, read));
+            return read;
+        }
+
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            FormatBytes(read: false, buffer.AsSpan(offset, count));
+            _stream.Write(buffer, offset, count);
+        }
+
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
```

Comment:
Shouldn't we override the other overloads as well? e.g. the Memory<byte> one? I guess we don't use these in HttpConnection/Http2Connection, but it seems like we might in the future.

Reply:
We can; it doesn't matter functionally. As this is inefficient logging for testing purposes, I didn't bother.

Reply:
I'm thinking that
(a) We might change the SocketsHttpHandler code to use a different overload in the future, and if we did, it would unexpectedly cause this test to break, which is kinda weird
(b) This logging functionality seems super useful in general, and I could easily see us using it elsewhere; so it seems worthwhile to invest a bit in making it comprehensive

Thoughts?

Reply:
> it would unexpectedly cause this test to break

Why?